### PR TITLE
Sort by distance

### PR DESCRIPTION
**Pull request by seanlawrenz**

### CODE REVIEW

**Comment by PhilNorman2** [comment]
Hi Sean.  I was looking at this and it looks like you might pulling the estimated_distance from the Agency JSON object.  Is that true?  The idea was to use the estimated_distance from the Event JSON because in a very small minority of cases the address of the Agency and Event differ.  I am afraid I confused things by putting the estimated_distance in both objects because it might be useful for other purposes down the road.   It probably doesn't matter at this stage but I think this is something David should weigh in on  (if true).

File: src/Utils/__test__/EventHandler.test.js [hunk 1]
```diff
@@ -158,14 +164,45 @@ test('should sort events into dates', () => {
...
 });
 
-test(`should return a final object sorted by events_date's date`, () => {
-  const shouldBeFirst = { ...mockAgency, events: [{ ...mockEvent , event_dates: [{ ...mockEventDate, date: '2020-04-01' }] }] };
-  const shouldBeSecond = { ...mockAgency, events: [{ ...mockEvent , event_dates: [{ ...mockEventDate, date: '2020-04-15' }] }] };
-  const testData = [shouldBeSecond, shouldBeFirst];
+test(`should return a final object sorted by events_date's date and distance`, () => {
+  const shouldBeFirst = {
+    ...mockAgency,
+    name: "should be first",
```
[the remark this answers is not in the document]
Comment (davidpickering) [reply src/Utils/__test__/EventHandler.test.js:170]: @maheshkurma if you can double check those points I mentioned and confirm the desired behavior, we are good to approve and merge.

**Comment by davidpickering** [comment]
> 
> 
> Hi Sean. I was looking at this and it looks like you might pulling the estimated_distance from the Agency JSON object. Is that true? The idea was to use the estimated_distance from the Event JSON because in a very small minority of cases the address of the Agency and Event differ. I am afraid I confused things by putting the estimated_distance in both objects because it might be useful for other purposes down the road. It probably doesn't matter at this stage but I think this is something David should weigh in on (if true).

Great point @PhilNorman2 . Yes, for the event_date cards, we want to use the estimated_distance value from the event (parent object of the event_date). Like Phil said, for most agencies they are the same, but when they are different, it really matters.  Also, I'm sorry I don't understand the source code well enough to provide better feedback - this might be what is already written?

**Comment by maheshkurma** [comment]
@davidpickering  sort by distance (estimated_distance) is perfectly working fine (shortest to longest), but it is not handling null cases., i tried to test locally
example: if estimated_distance is null (" ") then it is not showing at the end of the list (within the same day).,
To Handle null cases i changed line from "sort((a,b) => a.estimated_distance - b.estimated_distance);" to ".sort((a,b) => (a.estimated_distance? a.estimated_distance: Infinity) - (b.estimated_distance?b.estimated_distance:Infinity));" in file src/Utils/EventHandler.js  it works for me including null cases.

**Comment by davidpickering** [comment]
> 
> 
> @davidpickering @seanlawrenz sort by distance (estimated_distance) is perfectly working fine (shortest to longest), but it is not handling null cases., i tried to test locally
> example: if estimated_distance is null (" ") then it is not showing at the end of the list (within the same day).,
> To Handle null cases i changed line from "sort((a,b) => a.estimated_distance - b.estimated_distance);" to ".sort((a,b) => (a.estimated_distance? a.estimated_distance: Infinity) - (b.estimated_distance?b.estimated_distance:Infinity));" in file (src/Utils/EventHandler.js) it works for me including null cases.

@maheshkurma thank you for running that down. That seems like a logical approach, and I think we have the desired result.

Also, I'm sorry for my poor choice of words. But @PhilNorman2 reminded me he had contained really well for NULL in the results, and had replaced with an empty string. Full details are on his [PR3](https://github.com/midohiofoodbank/freshtrak-pantry-finder-api/pull/3). I believe you recognized this and modified the code appropriately to handle empty strings instead of NULL. 👍

**Comment by PhilNorman2** [comment]
@maheshkurma, would you provide an Agencies API query that is returning empty strings, "" for estimated_distance in the response?  I just want to make sure there is not a problem in the API.  It should not return "" for estimated_distance if the zip_code parameter is sent in with the request.

**Comment by maheshkurma** [comment]
@PhilNorman2 actually YES there is no such data(agencies) that returning empty strings, "" for estimated_distance in the response. API is working perfect.
I'm just trying to handle if estimated_distance returns " " that should show at the end of the list (within the same day).
I forcefully edited the JSON response(estimated_distance  " ") and tested in UI....
Actually you are right that make sense if there is no such scenario why are we handling it, it would be just an extra check...

**Comment by PhilNorman2** [comment]
>Great point @PhilNorman2 . Yes, for the event_date cards, we want to use the estimated_distance value from the event (parent object of the event_date). Like Phil said, for most agencies they are the same, but when they are different, it really matters. Also, I'm sorry I don't understand the source code well enough to provide better feedback - this might be what is already written?

@davidpickering, @seanlawrenz, @maheshkurma, I didn't see an answer to this question.  Is the code using the estimated_distance in the Event JSON object (correct) of the estimated_distance in the Agencies JSON object (incorrect).

**Comment by maheshkurma** [comment]
this code is using the estimated_distance in the Event JSON object.

**Comment by seanlawrenz** [comment]
Re requesting review. Thanks @maheshkurma for catching the `null` case. 💯

**Pull request merged**